### ISSUE TITLE
[Bug]: delete bug

### ISSUE THREAD

**Issue by fengttt** [matrixorigin/matrixone#11963]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93): main
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

```
mysql> create table dt (i int not null primary key, j int, t text);
mysql> insert into dt select result, result + 1, 'foobarzoo' || result from generate_series(1, 10000000) t;
mysql> select count(*) from dt;
+----------+
| count(*) |
+----------+
| 10000000 |
+----------+
1 row in set (0.06 sec)

mysql> delete from dt where i %2 = 0;
Query OK, 5000000 rows affected (14.47 sec)

mysql> select count(*) from dt;
+----------+
| count(*) |
+----------+
|  5000000 |
+----------+
1 row in set (0.24 sec)

mysql> delete from dt where j %2 = 0;
Query OK, 5000000 rows affected (1 min 0.85 sec)

mysql> select count(*) from dt;
+----------+
| count(*) |
+----------+
|  5000000 |
+----------+
1 row in set (0.43 sec)
```

All data should be deleted.   Redo and check
```
mysql> select * from dt where j %2 = 0 limit 2;
+---------+---------+------------------+
| i       | j       | t                |
+---------+---------+------------------+
| 6291457 | 6291458 | foobarzoo6291457 |
| 6291459 | 6291460 | foobarzoo6291459 |
+---------+---------+------------------+
2 rows in set (0.32 sec)

mysql> delete from dt where j %2 = 0;
ERROR 20619 (HY000): w-w conflict
mysql> select count(*) from dt;
+----------+
| count(*) |
+----------+
|  5000000 |
+----------+
1 row in set (0.23 sec)

mysql> delete from dt where j %2 = 0;
ERROR 20619 (HY000): w-w conflict
```

### Expected Behavior

The second delete should delete 5000000 rows.   We did not delete those.

Repeat this may cause w-w conflict.

**Comment by fengttt** [comment]
mysql> delete from dt;
Query OK, 5000000 rows affected (13.22 sec)

mysql> select count(*) from dt;
+----------+
| count(*) |
+----------+
|  5000000 |
+----------+
1 row in set (0.21 sec)

mysql>

**Comment by fengttt** [comment]
Complete, simple repro here.    The second delete, is both very slow and wrong.

```
mysql> create table dt (i int not null primary key, j int, t text);
Query OK, 0 rows affected (0.02 sec)

mysql> insert into dt select result, result + 1, 'foobarzoo' || result from generate_series(1, 10000000) t;
Query OK, 10000000 rows affected (9.40 sec)

mysql> delete from dt;
Query OK, 10000000 rows affected (16.72 sec)

mysql> select count(*) from dt;
+----------+
| count(*) |
+----------+
|        0 |
+----------+
1 row in set (0.13 sec)

mysql> insert into dt select result, result + 1, 'foobarzoo' || result from generate_series(1, 10000000) t;
Query OK, 10000000 rows affected (8.53 sec)

mysql> select count(*) from dt;
+----------+
| count(*) |
+----------+
| 10000000 |
+----------+
1 row in set (0.06 sec)

mysql> delete from dt where i % 2 = 0;
Query OK, 5000000 rows affected (13.78 sec)

mysql> select count(*) from dt;
+----------+
| count(*) |
+----------+
|  5000000 |
+----------+
1 row in set (0.25 sec)

mysql> delete from dt;
Query OK, 5000000 rows affected (1 min 0.72 sec)

mysql> select count(*) from dt;
+----------+
| count(*) |
+----------+
|  5000000 |
+----------+
1 row in set (0.55 sec)

mysql>
```

**Comment by XuPeng-SH** [comment]
fixed by #11969

**Comment by heni02** [comment]
confirm,closed
commit id:9b9e8091237333f80e3b828462a5ac31c8b083bf
![image](https://github.com/matrixorigin/matrixone/assets/113406637/2c8941c5-2124-47a1-860c-0e14e453a178)
